### PR TITLE
Vectorize Keccak state copy, remove temp array use

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Vectorized Xor to state
- Removed additional temp stackalloc array + copy
- Simplified
- Fast copy to output for 256 and 512 bit sizes

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs [hunk 1]
```diff
@@ -356,47 +356,102 @@ public static uint[] ComputeBytesToUint(ReadOnlySpan<byte> input, int size)
...
+                stateBytes[..output.Length].CopyTo(output);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void XorVectors(Span<byte> state, ReadOnlySpan<byte> input)
+        {
+            int length = input.Length;
+            int i = 0;
+            if (Vector256<byte>.IsSupported)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs:420]: Where `Vector512`?

File: src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs [hunk 2]
```diff
@@ -356,47 +356,102 @@ public static uint[] ComputeBytesToUint(ReadOnlySpan<byte> input, int size)
...
+                // Apply padding within the current block
+                stateBytes[remainingInputLength] ^= 0x01; // Append bit '1' after the input
+                stateBytes[roundSize - 1] ^= 0x80;        // Set the last bit of the block to '1'
             }
 
+            // Process the final block
             KeccakF(state);
-            MemoryMarshal.AsBytes(state[..(size / sizeof(ulong))]).CopyTo(output);
+
+            if (output.Length == Vector256<byte>.Count)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs:397]: no `Vector128`?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs:397]: Not for copying out as hashes are only 256 and 512

File: src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs]: Added

**Pull request merged**